modin/config/envvars.py

```diff
@@ -145,10 +145,21 @@ class Memory(EnvironmentVariable, type=int):
 
 class NPartitions(EnvironmentVariable, type=int):
     """
-    How many partitions to use by default
+    How many partitions to use for a Modin DataFrame (along each axis)
     """
 
     varname = "MODIN_NPARTITIONS"
+    # This flag is used to detect whether NPartitions is default value or not
+    _is_default = True
```

Comment:
I would say this should have initial value of `False`:
```suggestion
    _is_default = False
```
and be reset to `True` only in `_get_default()` method, otherwise it would be impossible for a user to override the cluster-defined value by setting an environment variable.

Reply:
Yes, you are right but in order to get all cases working we need to reset `_is_default` to `False` in `put` because the following case might be possible:
```python
import modin.pandas as pd
df = pd.DataFrame(frame_data) # NPartitions will be caught from cluster resources 
from modin.config import NPartitions
NPartitions.put(some_value)
df2 = pd.DataFrame(frame_data2) # NPartitions will still be caught from cluster resources 
```
Thanks for notice!